### PR TITLE
Create a new semi-fungible token standard request

### PR DESCRIPTION
**Pull request by muren102**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-draft_sft.md [hunk 1]
```diff
@@ -0,0 +1,312 @@
+---
+eip: <to be assigned>
```
Comment (MicahZoltu) [inline EIPS/eip-draft_sft.md:2]:
```suggestion
eip: 4635
```
Note: Please also change the filename to `EIPS/eip-4635.md`.

Comment (muren102) [reply EIPS/eip-draft_sft.md]: I've updated the file with 4635. Thanks

**Comment by MicahZoltu** [comment]
The bot won't ping the appropriate people until you mark the PR as ready for review.

**Comment by muren102** [comment]
> The bot won't ping the appropriate people until you mark the PR as ready for review.

Hi Micah
Thanks for your reminder.
I've set the PR as ready for review. 
Anyone else should I invite for the review? thanks

File: EIPS/eip-4635.md [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+        This function MUST revert if it rejects the transfer(s).
+        Return of any other value than the prescribed keccak256 generated value MUST result in the transaction being reverted by the caller.
+        @param _operator  The address which initiated the batch transfer (i.e. msg.sender)
+        @param _from      The address which previously owned the token
+        @param _types     An array containing token types being transferred (order and length must match _values array)
+        @param _values    An array containing amounts of each token being transferred (order and length must match _ids array)
+        @param _data      Additional data with no specified format
+        @return           `bytes4(keccak256("onERCSFTBatchReceived(address,address,uint256[],uint256[],bytes)"))`
+    */
+    function onERCSFTBatchReceived(address _operator, address _from, uint256[] calldata _types, uint256[] calldata _values, bytes calldata _data) external returns(bytes4);
```
Comment (linvictor88) [inline EIPS/eip-4635.md:243]: replace with number

**Review by linvictor88 — COMMENTED** [review]
change SFT to corresponding number as well as in coding implemenation

File: EIPS/eip-4635.md (hunk 2 above)
Comment (muren102) [reply EIPS/eip-4635.md:243]: I've added the number 0x198fb191 in "This function MUST return `bytes4(keccak256("onERCSFTReceived(address,address,uint256,uint256,bytes)"))` (i.e. 0x198fb191) if it accepts the transfer." and 0xa3a64bf1 in "This function MUST return `bytes4(keccak256("onERCSFTBatchReceived(address,address,uint256[],uint256[],bytes)"))` (i.e. 0xa3a64bf1) if it accepts the transfer(s)." I will also add the number in the return

File: EIPS/eip-4635.md [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+author: Bo Lin (@linvictor88), Danyang Li (@muren102), Frank Shen (@d22shen)
+discussions-to: https://ethereum-magicians.org/t/eip-4635-semi-fungible-token-standard/7923
+status: Draft
+type: Standards Track
+category: ERC
+created: 2021-12-16
+requires: 721
+---
+
+## Simple Summary
```
Comment (lightclient) [inline EIPS/eip-4635.md:13]: This field has been deprecated in favor of the `description` preamble field. Please update accordingly.

File: EIPS/eip-4635.md [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+A standard interface for contracts that manage semi-fungible token(s). A single deployed contract usually includes multiple types of fungible tokens and one single transformational non-fungible token.
+
+## Abstract
+This standard outlines a smart contract interface that can represent semi-fungible tokens (SFT). SFTs start out as types of fungible tokens and finally end as non-fungible tokens to provide value. ERC-1155 can manage combination of fungible tokens and non-fungible tokens, however, it's intractable for semi-fungible token representation. So this standard is specially for semi-fungible token use, in which fungible token types are managed like ERC-1155 and non-fungible tokens are totally managed by ERC-721.
```
Comment (lightclient) [inline EIPS/eip-4635.md:17]: It's not clear to me what the term "semi-fungible" means here?

File: EIPS/eip-4635.md [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+**Standards**
+- [ERC-721 Non-Fungible Token Standard](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-721.md)
+- [ERC-165 Standard Interface Detection](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-165.md)
+- [ERC-1155 Multi Token Standard](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1155.md)
+
+**Implementations**
+- [ERC-SFT Reference Implementation](https://github.com/linvictor88/ERC-SFT)
+
+## Security Considerations
+All EIPs must contain a section that discusses the security implications/considerations relevant to the proposed change. Include information that might be important for security discussions, surfaces risks and can be used throughout the life cycle of the proposal. E.g. include security-relevant design decisions, concerns, important discussions, implementation-specific guidance and pitfalls, an outline of threats and risks and how they are being addressed. EIP submissions missing the "Security Considerations" section will be rejected. An EIP cannot proceed to status "Final" without a Security Considerations discussion deemed sufficient by the reviewers.
```
Comment (lightclient) [inline EIPS/eip-4635.md:309]: Please remove this text and describe any security considerations regarding the EIP.

**Review by lightclient — CHANGES_REQUESTED** [review]
Left a few comments - I think I generally am not understanding the concept of "semi-fungibility" and how this differs from EIP-1155. The prose is a bit difficult to understand, I recommend spending some time to revise your abstract and motivation sections. Thanks.

File: EIPS/eip-4635.md (hunk 4 above)
Comment (muren102) [reply EIPS/eip-4635.md:17]: SFTs are a relatively new group of tokens that can be both fungible and non-fungible during their lifecycle. 
e.g. the art museum tickets, it is FT in the beginning. and when it is bought by someone, the ticket will be updated with specific ID and name and now it is NFT.

File: EIPS/eip-4635.md (hunk 5 above)
Comment (muren102) [reply EIPS/eip-4635.md:309]: OK. Thanks.

File: EIPS/eip-4635.md (hunk 3 above)
Comment (muren102) [reply EIPS/eip-4635.md:13]: OK. thanks

**Pull request closed without merging**